src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/CreateClaimSpecCallbackHandler.java

```diff
@@ -439,6 +440,17 @@ private CallbackResponse validateRespondent1Address(CallbackParams callbackParam
 
     }
 
+    private CallbackResponse validateRespondent2Address(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        List<String> errors = postcodeValidator.validatePostCodeForDefendant(
+            caseData.getRespondent2().getPrimaryAddress().getPostCode());
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .errors(errors)
+            .build();
+
```

Comment:
since the address validation is going to be the same for both respondent 1 and 2, I think it's worth it to move that code to another method that receives the respondent and then checks for errors, so that you have

private List<String> validateRespondentAddress(Party respondent) {
  return postcodeValidator.validatePostCodeForDefendant(respondent.primaryAddress.postCode);
}

private CallbackResponse validateRespondent1Address(CallbackParams params) {
  CaseData caseData = callbackParams.getCaseData();
  List<String> errors = validateRespondentAddress(caseData.respondent1);
  
  return AboutToStartOrSubmitCallbackResponse.builder().errors(errors).build;
}

private CallbackResponse validateRespondent2Address(CallbackParams params) {
  CaseData caseData = callbackParams.getCaseData();
  List<String> errors = validateRespondentAddress(caseData.respondent2);
  
  return AboutToStartOrSubmitCallbackResponse.builder().errors(errors).build;
}

Or you can move every common thing to the new method and use Function to get the right respondent:
private List<String> validateRespondentAddress(CallbackParams params, Function<CaseData,Party> getRespondent) {
    CaseData caseData = callbackParams.getCaseData();
  List<String> errors = postcodeValidator.validatePostCodeForDefendant(getRespondent.apply(caseData).primaryAddress.postCode);
  
  return AboutToStartOrSubmitCallbackResponse.builder().errors(errors).build;
}

private CallbackResponse validateRespondent1Address(CallbackParams params) {
  return validateRespondentAddress(params, CaseData::getRespondent1);
}

private CallbackResponse validateRespondent2Address(CallbackParams params) {
  return validateRespondentAddress(params, CaseData::getRespondent2);
}

Reply:
agree, updated it in next commit